gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagProcessingEngineMetrics.java

```diff
@@ -0,0 +1,155 @@
+    switch (metricName) {
+      case ServiceMetricNames.DAG_ACTIONS_STORED:
+        updateMetricForDagAction(this.dagActionsStoredMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTIONS_OBSERVED:
+        updateMetricForDagAction(this.dagActionsObservedMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTIONS_LEASES_OBTAINED:
+        updateMetricForDagAction(this.dagActionsLeasesObtainedMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTIONS_NO_LONGER_LEASING:
+        updateMetricForDagAction(this.dagActionsNoLongerLeasingMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTION_LEASE_REMINDERS_SCHEDULED:
+        updateMetricForDagAction(this.dagActionLeaseRemindersScheduledMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTION_REMINDERS_PROCESSED:
+        updateMetricForDagAction(this.dagActionRemindersProcessedMeters, dagActionType);
+        break;
+      // TODO: implement evaluating max retries later
+        case ServiceMetricNames.DAG_ACTIONS_EXCEEDED_MAX_RETRY:
+        updateMetricForDagAction(this.dagActionsExceededMaxRetryMeters, dagActionType);
+          break;
+      case ServiceMetricNames.DAG_ACTIONS_INIT_FAILED:
+        updateMetricForDagAction(this.dagActionsInitFailedMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTIONS_INIT_SUCCEEDED:
+        updateMetricForDagAction(this.dagActionsInitSucceededMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTION_EXECUTIONS_FAILED:
+        updateMetricForDagAction(this.dagActionExecutionsFailedMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTION_EXECUTIONS_SUCCEEDED:
+        updateMetricForDagAction(this.dagActionExecutionsSucceededMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTION_CONCLUSIONS_FAILED:
+        updateMetricForDagAction(this.dagActionConclusionsFailedMeters, dagActionType);
+        break;
+      case ServiceMetricNames.DAG_ACTION_CONCLUSIONS_SUCCEEDED:
+        updateMetricForDagAction(this.dagActionConclusionsSucceededMeters, dagActionType);
+        break;
+      default:
+        log.warn("Skipping marking metric because no meter found to match it. Metric name: {}", metricName);
+        break;
+    }
+  }
+
+  /**
+   * Generic helper used to increment a metric corresponding to the dagActionType in the provided map. It assumes the
+   * meter for each dagActionType can be identified by its name.
+   */
+  private void updateMetricForDagAction(ConcurrentMap<String, ContextAwareMeter> metricMap,
+      DagActionStore.DagActionType dagActionType) {
+      if (metricMap.containsKey(dagActionType.toString())) {
+        metricMap.get(dagActionType.toString()).mark();
+      } else {
+        log.warn("Skipping metric. No meter exists for dagActionType {} in metricsMap {}");
+      }
```

Comment:
this feels like a massive "can't happen" situation (given the strong typing of `dagActionType`.  therefore, it either ought to throw an exception or do a JIT initialization of the `metricMap` for the new type